src/uucore/src/lib/features/selinux.rs

```diff
@@ -6,22 +6,34 @@
 use std::path::Path;
 
 use selinux::SecurityContext;
+use thiserror::Error;
 
-#[derive(Debug)]
-pub enum Error {
+#[derive(Debug, Error)]
+pub enum SeLinuxError {
+    #[error("SELinux is not enabled on this system")]
     SELinuxNotEnabled,
-    FileOpenFailure,
-    ContextRetrievalFailure,
-    ContextConversionFailure,
+
+    #[error("Failed to open the file: {0}")]
+    FileOpenFailure(String),
+
+    #[error("Failed to retrieve the security context: {0}")]
+    ContextRetrievalFailure(String),
+
+    #[error("failed to set default file creation context to '{0}': {1}")]
+    ContextSetFailure(String, String),
+
+    #[error("failed to set default file creation context to '{0}': {1}")]
+    ContextConversionFailure(String, String),
```

Comment:
For consistency reasons I would start all error messages with either `Failed` or `failed`.